jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/command/AddDataCaseFileInstanceCommand.java

```diff
@@ -73,6 +74,15 @@ public Void execute(Context context) {
         
         KieSession ksession = ((RegistryContext) context).lookup( KieSession.class );
         
+        org.jbpm.process.instance.ProcessInstance pi = (org.jbpm.process.instance.ProcessInstance) ksession.getProcessInstance(processInstanceId);
+        VariableScope variableScope = (VariableScope) pi.getContextContainer().getDefaultContext(VariableScope.VARIABLE_SCOPE);
+        
+        for (String name: parameters.keySet())
+        {
+            if (variableScope.isReadOnly(VariableScope.CASE_FILE_PREFIX+name))
```

Comment:
It should control also that variable has a previous value assigned which tries to be overwritten

Reply:
Done